### PR TITLE
Allow recursive compute with proxy accessor

### PR DESCRIPTION
**Pull request by kxl-adsk**
The proxy accessor is allowing DG dependencies between Maya's data model and USD. This means we can connect input from a DG node to proxy shape to drive USD attributes AND we can connect the proxy shape outputs to make USD attributes drive other DG nodes.

This also allows injecting DG dependencies between two USD attributes, like for example creating parent constraint between two prims. In order to compute such a setup correctly, we need to allow nested / recursive compute with the proxy accessor.

This PR is doing exactly that and an example of such DG setup is provided as a regression test.

### CODE REVIEW

File: lib/mayaUsd/base/id.h [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+#ifndef MAYAUSD_ID_H
+#define MAYAUSD_ID_H
+
+#include <climits>
+#include <string>
+
+namespace MAYAUSD_NS_DEF {
+
+/*! \class Id
```
Comment (kxl-adsk) [inline lib/mayaUsd/base/id.h:25]: We have been starting to use "version" numbers in base proxy shape and VP2RenderDelegate. Tracking what different int's and long's mean can be tricky, so we are bringing this little class to solve it.

File: lib/usd/pxrUsdPreviewSurface/usdPreviewSurface.cpp [hunk 2]
```diff
@@ -425,8 +424,8 @@ MStatus PxrMayaUsdPreviewSurface::compute(const MPlug& plug, MDataBlock& dataBlo
         CHECK_MSTATUS(status);
         float opacity = opacityData.asFloat();
 
-        MObject opacityThresholdAttr
-            = depNodeFn.attribute(PxrMayaUsdPreviewSurfaceTokens->OpacityThresholdAttrName.GetText());
+        MObject opacityThresholdAttr = depNodeFn.attribute(
```
Comment (kxl-adsk) [inline lib/usd/pxrUsdPreviewSurface/usdPreviewSurface.cpp:427]: Only formatting change from clang-format

File: test/lib/CMakeLists.txt [hunk 3]
```diff
@@ -5,12 +5,12 @@ set(TEST_SCRIPT_FILES
     testMayaUsdConverter.py
     testMayaUsdPythonImport.py
     testMayaUsdLayerEditorCommands.py
+    testMayaUsdCacheId.py
 )
 
-if (MAYA_APP_VERSION VERSION_GREATER 2020)
+if (UFE_FOUND AND MAYA_APP_VERSION VERSION_GREATER 2020)
```
Comment (kxl-adsk) [inline test/lib/CMakeLists.txt:11]: Proxy accessor test requires UFE.

File: test/lib/testMayaUsdProxyAccessor.py [hunk 4]
```diff
@@ -53,6 +53,7 @@ def setUpClass(cls):
         """
         if not cls.pluginsLoaded:
             cls.pluginsLoaded = mayaUtils.isMayaUsdPluginLoaded()
+            cmds.loadPlugin('matrixNodes')
```
Comment (kxl-adsk) [inline test/lib/testMayaUsdProxyAccessor.py:56]: For inverseMatrix node

File: lib/mayaUsd/nodes/proxyAccessor.cpp [hunk 5]
```diff
@@ -50,6 +51,94 @@
 
 namespace MAYAUSD_NS_DEF {
 
+/*! /brief  Scoped object setting up compute context for accessor
+
+    Proxy accessor supports nested compute that allows injecting DG dependencies to USD. More
```
Comment (williamkrick) [inline lib/mayaUsd/nodes/proxyAccessor.cpp:56]: Is there an extra space before "Proxy"?

Comment (williamkrick) [reply lib/mayaUsd/nodes/proxyAccessor.cpp:56]: Maybe the other lines have only 3 spaces.

File: lib/mayaUsd/nodes/proxyAccessor.cpp [hunk 6]
```diff
@@ -50,6 +51,94 @@
...
+        _proxyInclusiveMatrix = proxyDagPath.inclusiveMatrix();
+
+        // Only increment evaluation ID for top most evaluation scope.
+        // Nested scope is allowed, but shouldn't in really be needed.
+        if (!_restoreState) {
+            accessor._evaluationId.next();
+        }
+    }
+
+    //! \brief  Construct compute context for both inputs
```
Comment (williamkrick) [inline lib/mayaUsd/nodes/proxyAccessor.cpp:92]: This comment cut off?

**Review by williamkrick — APPROVED** [review]

File: lib/mayaUsd/nodes/proxyAccessor.cpp (hunk 5 above)
Comment (kxl-adsk) [reply lib/mayaUsd/nodes/proxyAccessor.cpp:56]: Good catch!

File: lib/mayaUsd/nodes/proxyAccessor.cpp (hunk 6 above)
Comment (kxl-adsk) [reply lib/mayaUsd/nodes/proxyAccessor.cpp]: Fixed.

**Review by williamkrick — APPROVED** [review]

**Pull request merged**